src/gsy_e/models/strategy/energy_parameters/energy_params_eb.py

```diff
@@ -21,37 +21,190 @@
+            self, energy_kWh: float, market_slot: pendulum.DateTime,
+            product_type: AvailableMarketTypes):
+        """
+        When a trade happens, we want to split the energy through the entire year following a
+        standard solar profile.
+
+        Args:
+            energy_kWh: the traded energy in kWh.
+            market_slot: the slot targeted by the trade. For forward exchanges, this represents
+                the entire period of time over which the trade profile will be generated.
+            product_type: One of the available market types.
+        """
+
+    @property
+    @abstractmethod
+    def state(self) -> "StateInterface":
+        """Retrieve the internal state of the asset."""
+
+    @staticmethod
+    def day_forward_slots(market_slot: pendulum.DateTime):
+        """Get the market slots for the day forward market."""
+        return [
+            market_slot,
+            market_slot + pendulum.duration(minutes=15),
+            market_slot + pendulum.duration(minutes=30),
+            market_slot + pendulum.duration(minutes=45)]
+
+
+class ConsumptionStandardProfileEnergyParameters(ForwardEnergyParams):
     """Energy parameters of the load strategy for the forward markets."""
 
     def __init__(self, capacity_kW):
+        super().__init__()
         self.capacity_kW: float = capacity_kW
 
         self._state = LoadState()
         self._area = None
         self._profile_generator: Optional[ForwardTradeProfileGenerator] = None
 
+    @property
+    def state(self) -> LoadState:
+        """Retrieve the internal state of the asset."""
+        return self._state
+
     def serialize(self):
         """Return dict with the current energy parameter values."""
         return {
             "capacity_kW": self.capacity_kW
         }
 
+    @property
+    def capacity_kWh(self):
+        """Capacity of the load in kWh."""
+        return convert_kW_to_kWh(self.capacity_kW, self._area.config.slot_length)
+
+    def get_available_energy_kWh(
+            self, market_slot: pendulum.DateTime, market_type: AvailableMarketTypes):
+        """Get the available bid energy of the load."""
+        if market_type == AvailableMarketTypes.INTRADAY:
+            return self._state.get_energy_requirement_Wh(market_slot) / 1000.0
+        if market_type == AvailableMarketTypes.DAY_FORWARD:
+            return min(
+                self._state.get_energy_requirement_Wh(slot) / 1000.0
+                for slot in self.day_forward_slots(market_slot)
+            )
+
+        reference_slot = market_slot.set(hour=12, minute=0, tz=pendulum.UTC)
+
+        if self._state.get_desired_energy_Wh(reference_slot) <= FLOATING_POINT_TOLERANCE:
+            scaling_factor = 0.
+        else:
+            scaling_factor = (
+                    self._state.get_energy_requirement_Wh(reference_slot) /
+                    self._state.get_desired_energy_Wh(reference_slot))
+        return abs(scaling_factor) * self.capacity_kWh
```

Comment:
Could you please explain this part? What's the "available energy" for a load? The docstring says:
> Get the available bid energy of the load

but it sounds a bit counterintuitive to me to grasp. Is the "available bid energy" the same thing as "required energy"? If so, could we use the same "energy requirement" naming we used in the state?

Also, regarding the scaling factor. AFAIU, this means something like:

$\displaystyle scaling\ factor = \frac{energy\ still\ needed}{original\ total\ energy\ needed}$

Is this correct? What do we obtain by multiplying this for the capacity?

Reply:
I added a general comment that explains it already, however let me also amend the docstring in order to better reflect the purpose of this method. To recap, available energy == energy that the asset can generate / consumed , that has not been traded yet. Therefore the formula for the scaling factor is the following:
`scaling factor = not-traded-energy-at-12:00 / total-energy-needed-at-12:00` 

As for why it is needed, the reason is that we always use the peak energy as a unit for orders. The peak energy though is not necessarily at 12:00 on every profile. In order to avoid iterating through the profile and finding the peak energy slot, we instead use a timeslot that we know has nonzero energy (12:00 is one of these), we calculate the scaling factor at 12:00, and then we use this scaling factor in order to scale the peak value of the profile (which we know is self.peak_energy_kWh), which we can then use to post orders (since the scaling factor has removed from the peak the energy that has already been traded). Hope that the above is clear, let me know if further explanations are needed here!